include/grpc/impl/codegen/grpc_types.h

```diff
@@ -615,6 +616,9 @@ typedef struct {
   /** If non-NULL, will be set to point to a string containing the
    * service config used by the channel in JSON form. */
   char** service_config_json;
+  /** If non-NULL, Will be set to the channels connectivity state
```

Comment:
In order for the "if non-null" semantic to work, the field needs to be a pointer.